gpt4all-chat/download.cpp

```diff
@@ -199,20 +199,27 @@ void Download::removeModel(const QString &modelFile)
         incompleteFile.remove();
     }
 
+    bool shouldRemoveInstalled = false;
     QFile file(filePath);
     if (file.exists()) {
         const ModelInfo info = ModelList::globalInstance()->modelInfoByFilename(modelFile);
-        ModelList::globalInstance()->removeInstalled(info);
+        shouldRemoveInstalled = info.installed && !info.isClone() && (info.isDiscovered() || info.description() == "" /*indicates sideloaded*/);
+        if (shouldRemoveInstalled)
+            ModelList::globalInstance()->removeInstalled(info);
         Network::globalInstance()->sendRemoveModel(modelFile);
         file.remove();
     }
 
-    ModelList::globalInstance()->updateDataByFilename(modelFile, ModelList::InstalledRole, false);
-    ModelList::globalInstance()->updateDataByFilename(modelFile, ModelList::BytesReceivedRole, 0);
-    ModelList::globalInstance()->updateDataByFilename(modelFile, ModelList::BytesTotalRole, 0);
-    ModelList::globalInstance()->updateDataByFilename(modelFile, ModelList::TimestampRole, 0);
-    ModelList::globalInstance()->updateDataByFilename(modelFile, ModelList::SpeedRole, QString());
-    ModelList::globalInstance()->updateDataByFilename(modelFile, ModelList::DownloadErrorRole, QString());
+    if (!shouldRemoveInstalled) {
+        QVector<QPair<int, QVariant>> data;
+        data.append({ ModelList::InstalledRole, false });
+        data.append({ ModelList::BytesReceivedRole, 0 });
+        data.append({ ModelList::BytesTotalRole, 0 });
+        data.append({ ModelList::TimestampRole, 0 });
+        data.append({ ModelList::SpeedRole, QString() });
+        data.append({ ModelList::DownloadErrorRole, QString() });
```

Comment:
You could also use brace initialization in places like this:
```suggestion
        QVector<QPair<int, QVariant>> data {
            { ModelList::InstalledRole, false },
            { ModelList::BytesReceivedRole, 0 },
            { ModelList::BytesTotalRole, 0 },
            { ModelList::TimestampRole, 0 },
            { ModelList::SpeedRole, QString() },
            { ModelList::DownloadErrorRole, QString() },
        };
```